File: 2022/test/day_01_test.exs
defmodule AoC.Day01Test do
  use ExUnit.Case
  doctest AoC.Day01

  alias AoC.Day01

  describe "part 1" do
    test "sample 1" do
      input = "1000
2000
3000

4000

5000
6000

7000
8000
9000

10000"
      assert Day01.part_1(input) == 24_000
    end
  end

  describe "part 2" do
    test "sample 1" do
      input = "1000
2000
3000

4000

5000
6000

7000
8000
9000

10000"
      assert Day01.part_2(input) == 45_000
    end
  end
end

File: 2023/src/days/__tests__/day01.test.ts
import DayModule from '@/days/day01';

describe('part 1', () => {
  test('sample', () => {
    const input = `1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet`;

    expect(DayModule.part1(input)).toBe(142);
  });
});

describe('part 2', () => {
  test('sample', () => {
    const input = `two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen`;

    expect(DayModule.part2(input)).toBe(281);
  });
});
